packages/core-contracts/src/contracts/FleetProxy.sol

```diff
@@ -0,0 +1,334 @@
+import {ReentrancyGuard} from "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import {Pausable} from "@openzeppelin/contracts/utils/Pausable.sol";
+import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import {ProtocolAccessManaged, ContractSpecificRoles} from "@summerfi/access-contracts/contracts/ProtocolAccessManaged.sol";
+import {IFleetCommander} from "../interfaces/IFleetCommander.sol";
+import {IFleetProxy} from "../interfaces/IFleetProxy.sol";
+import {ICrossChainAssetReceiver} from "@summerfi/chain-bridge/interfaces/ICrossChainAssetReceiver.sol";
+import {IInflightAssetTracking} from "@summerfi/chain-bridge/interfaces/IInflightAssetTracking.sol";
+import {ICrossChainRegistry} from "../interfaces/ICrossChainRegistry.sol";
+
+/**
+ * @title FleetProxy
+ * @author SummerFi
+ * @notice Proxy contract that receives and holds assets on a satellite chain on behalf of a source chain fleet
+ * @dev Implements ICrossChainReceiver to handle cross-chain messages
+ */
+contract FleetProxy is
+    IFleetProxy,
+    IInflightAssetTracking,
+    ProtocolAccessManaged,
+    ReentrancyGuard,
+    Pausable
+{
+    using SafeERC20 for IERC20;
+
+    /// @notice Relationship type constant for ARK-FLEET relationships
+    bytes32 private constant ARK_FLEET_RELATIONSHIP = keccak256("ARK_FLEET");
+
+    /*//////////////////////////////////////////////////////////////
+                                ERRORS
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Error thrown when bridge router address is invalid
+    error InvalidBridgeRouter();
+    /// @notice Error thrown when bridge queue address is invalid
+    error InvalidBridgeQueue();
+    /// @notice Error thrown when registry address is invalid
+    error InvalidRegistry();
+    /// @notice Error thrown when fleet contract address is invalid
+    error InvalidFleetContract();
+    /// @notice Error thrown when withdrawal failed
+    error WithdrawalFailed();
+    /// @notice Thrown when the caller is not authorized to perform the action.
+    error Unauthorized();
+
+    /*//////////////////////////////////////////////////////////////
+                            STATE VARIABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The bridge router used for cross-chain communication
+    IBridgeRouter public immutable bridgeRouter;
+
+    /// @notice The bridge queue used for queuing cross-chain transfers
+    IBridgeQueue public immutable bridgeQueue;
+
+    /// @notice The CrossChainRegistry contract for managing cross-chain relationships
+    ICrossChainRegistry public immutable crossChainRegistry;
+
+    /// @notice The address of the Fleet contract that this proxy covers
+    address public immutable fleetContract;
+
+    /// @notice Amount of withdrawal assets currently in-flight (being bridged back)
+    uint256 public inflightWithdrawals;
```

Comment:
the `inflightWithdrawals` are newver reset, they're `up only`

Reply:
i see its updated by the queue on asset transfer - but wht if there are multiple transfers queued from single fleetProxy - they will be overwritten. 
Maybe limit the withdrawals to one and pull the inflight assets from the queue / bridge based on id. 
The callback might get missed when reading the code - Like I did